packages/components/src/components/es-empty-state/es-empty-state.tsx

```diff
@@ -0,0 +1,34 @@
+import { Component, h, Host, Prop, type VNode } from '@stencil/core';
+import type { EmptyStateLayout } from './types';
+
+/**
+ * Display an empty state with an illustration and a message.
+ */
+@Component({
+    tag: 'es-empty-state',
+    styleUrl: 'es-empty-state.css',
+    shadow: true,
+})
+export class EmptyState {
+    /** The header of the empty state. */
+    @Prop() header!: string;
+    /** The body of the empty state. */
+    @Prop() body!: string | VNode;
+    /** The layout of the empty state. */
+    @Prop() layout: EmptyStateLayout = 'vertical';
+
+    render() {
+        return (
+            <Host class={this.layout}>
+                <div class={'inner'}>
```

Comment:
Lets add some `part`s to these, to allow for external syling.

`inner`
`text`
`title`
`body`